Allow user to customize `cpu_target` in `create_app`

DilumAluthge opened the pull request:
This allows the user to customize the CPU target when creating an app.

The default (and strongly recommended) value is `APP_CPU_TARGET`, but the user can override it.

Basically, this finishes what was started in #43 and #52

DilumAluthge committed:
Allow user to customize `cpu_target` in `create_app`

DilumAluthge commented:
Basically, this finishes what we started in https://github.com/KristofferC/PackageCompilerX.jl/pull/43 and https://github.com/KristofferC/PackageCompilerX.jl/pull/52

KristofferC commented:
What do you want to override it to?

DilumAluthge commented:
In my particular use case, I need to add `-cx16` to the generic target because of a virtualization environment that intercepts it. (I don’t have control over the virtualization settings, so I have to fix it on the sysimage side.)

DilumAluthge commented:
But most people should use the default target, i.e. we shouldn’t add `-cx16` to the default value.

KristofferC commented:
Alright, that makes sense. But this needs to be added to the docstring so that it is discoverable. What's the error message you get if you don't add it? Does Julia suggest it?

DilumAluthge committed:
Add `cpu_target` to the docstrings of `create_app` and `create_sysimage`

DilumAluthge commented:
> this needs to be added to the docstring so that it is discoverable. 

I’ve added `cpu_target` to the docstring of `create_app`. It was missing from the docstring of `create_sysimage`, so I also added it there.

> What's the error message you get if you don't add it? Does Julia suggest it?

Unfortunately, it’s just the `ERROR: Unable to find compatible target in system image` error, and Julia refuses to start.

KristofferC commented:
https://github.com/JuliaComputing/PackageCompilerX.jl/commit/cb8c720e896a35caf353375e64280f14f2dbbaea

couldn't push to this branch for some reason
